src/RockBridge.cpp

```diff
@@ -89,18 +89,22 @@ void RockBridge::worldCreated(std::string const& worldName)
         // Create and initialize one rock component for each gazebo model
 
         sdf::ElementPtr modelSDF = (*model_it)->GetSDF();
-        if(modelSDF->HasElement("plugin"))
+        if (modelSDF->HasElement("plugin"))
         {
             sdf::ElementPtr pluginElement = modelSDF->GetElement("plugin");
-            std::string pluginFilename = pluginElement->Get<std::string>("filename");
-            gzmsg <<"RockBridge: found plugin "<< pluginFilename << std::endl;
-            if(pluginFilename == "libgazebo_thruster.so")
-            {
-                ThrusterTask* model_task = new ThrusterTask();
-                model_task->setGazeboModel(world, (*model_it) );
-                setupTaskActivity(model_task);
+            while (pluginElement) {
+                std::string pluginFilename = pluginElement->Get<std::string>("filename");
+                gzmsg <<"RockBridge: found plugin "<< pluginFilename << std::endl;
+                if (pluginFilename == "libgazebo_thruster.so")
+                {
+                    ThrusterTask* thruster_task = new ThrusterTask();
+                    thruster_task->setGazeboModel(world, (*model_it), pluginElement);
+                    setupTaskActivity(thruster_task);
+                }
+                pluginElement = pluginElement->GetNextElement("plugin");
```

Comment:
No, but you could group them into separate thruster plugins if you wish to. One use-case of this is for people to get as many thruster tasks (on the rock side) as they have on their real system (which can mean one plugin per thruster). Not something I would personally do, but I am sure we can find people who disagree

The bottom line is that the SDF spec allows it, so either we should generate an error if someone does it, or we should support it.